middleware/logger.go

```diff
@@ -0,0 +1,21 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/rs/zerolog"
+)
+
+// Make a logging middleware handler with the configured logger injected.
+func MakeLoggingMiddlewareHandler(logger zerolog.Logger) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		handler := func(w http.ResponseWriter, r *http.Request) {
+			requestLogMessage := fmt.Sprintf("%s %s %s", r.RemoteAddr, r.URL, r.Method)
+			logger.Info().Msg(requestLogMessage)
+			next.ServeHTTP(w, r)
```

Comment:
Would we want an after of `Request Complete` - I'm on the fence. It's not super necessary but could make a request chain easier to follow to termination.

Also, should the logger be passed down in a context?

Reply:
The solution to have logs after the request were pretty involved; a lot of the solutions I saw involved wrapping the `http.ResponseWriter` in another struct, and that seemed pretty intense since I'm not sure yet what we'd even want to log there. 

I inject the logger into the server instead since I thought that made more sense, but maybe the context would be more useful.

Reply:
Can't you just put a logger.Info() after the next() for an after-log? It may not have any value, and not worth the effort now if we don't get value-add, since  if it ever becomes important it's easy for the person who sees value.

Thoughts on that - if we inject it into the server, we can't pass it out if we call something else that's not this server. That's not necessarily a problem, especially since our expected footprint is small, and moving it to context later is not a huge lift because it's high on the chain.